src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionUITests.swift

```diff
@@ -143,7 +143,9 @@ class ENAUITests_04_ExposureSubmissionUITests: XCTestCase {
 		
 		// This may fail in future because of the Thank You card. So, just check the home screen
 		XCTAssertTrue(app.collectionViews.buttons["AppStrings.Home.submitCardButton"].waitForExistence(timeout: .long))
-		waitForNonExistence(of: app.buttons["AppStrings.ExposureSubmission.primaryButton"], timeout: .long)
+
+		// Back to homescreen
+		XCTAssertTrue(app.collectionViews.buttons["AppStrings.Home.submitCardButton"].waitForExistence(timeout: .long))
```

Comment:
Puneet and I invented the `waitForNonExistence` because due to the new modal presentation I think this check will pass even if the submission flow is not dismissed.

Reply:
I just quickly checked, for example the test `test_SubmitTAN_CancelOnTestResultScreen()` has this check but never actually arrives back at the home screen.

Reply:
I know you both worked on this but I oversaw some major flaws in that design during my review.
That implementation never passed the thrown error to the test. All I got was a grey notification that the expectation was not triggered but the test passed green - bad thing.

Also design wise I think it's still very weird to check for the non-existance of something and not the existence of the thing that should be there. Not having something in the `app` hierarchy does not mean it's ok. The view could be broken and this check would result in a success, like a check for the non-existence of zombies ;)
![corona is fine](https://user-images.githubusercontent.com/149976/102463623-a3d06300-404b-11eb-8c71-9f4cbeb9e1ae.jpg)

About the `test_SubmitTAN_CancelOnTestResultScreen`, I wonder why the test passes then? If we really need this 'non-existence' thing, we could bring it back but then in a consistent way and not the old implementation. I'm happy to help with that.